packages/cli-platform-apple/src/commands/runCommand/getBuildSettings.ts

```diff
@@ -0,0 +1,74 @@
+import {CLIError, logger} from '@react-native-community/cli-tools';
+import {IOSProjectInfo} from '@react-native-community/cli-types';
+import chalk from 'chalk';
+import child_process from 'child_process';
+
+export async function getBuildSettings(
+  xcodeProject: IOSProjectInfo,
+  mode: string,
+  buildOutput: string,
+  scheme: string,
+  target?: string,
+) {
+  const buildSettings = child_process.execFileSync(
+    'xcodebuild',
+    [
+      xcodeProject.isWorkspace ? '-workspace' : '-project',
+      xcodeProject.name,
+      '-scheme',
+      scheme,
+      '-sdk',
+      getPlatformName(buildOutput),
+      '-configuration',
+      mode,
+      '-showBuildSettings',
+      '-json',
+    ],
+    {encoding: 'utf8'},
+  );
+
+  const settings = JSON.parse(buildSettings);
+
+  const targets = settings.map(
+    ({target: settingsTarget}: any) => settingsTarget,
```

Comment:
let's assign the type information (even most basic one at this point) for `settings` parsed from `buildSettings` so that we don't need to cast to any here